lib/helper/Puppeteer.js

```diff
@@ -1841,10 +1841,11 @@ class Puppeteer extends Helper {
 
       if (locator.isXPath()) {
         waiter = contextObject.waitForFunction((locator, text, $XPath) => {
+          eval($XPath); // eslint-disable-line no-eval
```

Comment:
wherefore did you use eval?

Reply:
If we pass $XPath function as is to the Puppeteers waitForFunction it will be resolved as undefined. The possible reason is that $XPath function doesn't exit in Puppeteers browser context, so we need to add $XPath to the scope. The solution can be convert function to string and use eval in the browsers context. Exactly the same approach is used in waitForDetached or waitForEnabled functions. So I decided that eval can be acceptable approach here.